resident/resident-service/src/main/java/io/mosip/resident/util/ResidentServiceRestClient.java

```diff
@@ -317,8 +320,13 @@ private HttpEntity<Object> setRequestHeader(Object requestType, MediaType mediaT
 				HttpEntity<Object> httpEntity = (HttpEntity<Object>) requestType;
 				HttpHeaders httpHeader = httpEntity.getHeaders();
 				for (String key : httpHeader.keySet()) {
-					if (!(headers.containsKey("Content-Type") && Objects.equals(key, "Content-Type")))
-						headers.add(key, httpHeader.get(key).get(0));
+					if (!(headers.containsKey("Content-Type") && Objects.equals(key, "Content-Type"))){
+						if(httpHeader.isEmpty() || httpHeader.get(key) == null || httpHeader.get(key).isEmpty()
+						|| (httpHeader.get(key) != null && httpHeader.get(key).size() == 0) ||

```

Comment:
Correct the condition as '(httpHeader.get(key) != null && !httpHeader.get(key).isEmpty())'

Reply:
Done